src/main/java/br/com/deveficiente/bolaoapi/shared/validator/ChampionshipTeamsValidator.java

```diff
@@ -0,0 +1,30 @@
+package br.com.deveficiente.bolaoapi.shared.validator;
+
+import br.com.deveficiente.bolaoapi.services.team.TeamRepository;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.Optional;
+import java.util.Set;
+
+public class ChampionshipTeamsValidator implements ConstraintValidator<ChampionshipTeams, Set<Long>> {
```

Comment:
Opa, mal minha aqui! Acho que no cartão ficou mal explicado! A ideia é que você valide a quantidade de times escolhidos para o campeonato X a quantidade de times estabelecida. Quero evitar que seja cadastrado um campeonato cuja quantidade de times é 10 mas sem 10 times selecionados. Foi mal!

Reply:
Acho que o problema foi a pista que deixei, não ficou clara.
Mas essa validation é justamente pra esse propósito.

Vou mudar o nome da annotation!

Reply:
Opa, lá na classe Championship tem o atributo totalTeams. Aqui no seu validador você está verifciando se todos os ids de times selecionados estão cadastrados. O que é válido, sem duvida. Inclusive você pode usar o método ```findAllById```. Caso o número de times retornados for diferentes do número de ids, tem alguém faltando. 

Aí entendo que precisa de outra validaçõa para verificar o totalTeams Vs times selecionados :).

Reply:
Que era justamente essa validação ``` championship.getTeams().size() != championship.getTotalTeams()```

Foi mal se estou me passando :(

Reply:
Isso, essa validação foi para esse novo validator, o retorno dele na verdade.
```java
return requestCount == databaseCount;
```
Eu faço a consulta pois vai que o usuário mandou um id inexistente (daí o sistema cadastraria um campeonato que n segue essa regra).

Vou substituir pelo findAll